teos/src/watcher.rs

```diff
@@ -449,25 +405,26 @@ impl Watcher {
     fn filter_breaches(
         &self,
         breaches: HashMap<Locator, Transaction>,
-    ) -> (
-        HashMap<UUID, Breach>,
-        HashMap<UUID, cryptography::DecryptingError>,
-    ) {
-        let mut valid_breaches = HashMap::new();
-        let mut invalid_breaches = HashMap::new();
+    ) -> (Vec<(UUID, UserId, Breach)>, Vec<UUID>) {
+        let mut valid_breaches = Vec::new();
+        let mut invalid_breaches = Vec::new();
 
         // A cache of the already decrypted blobs so replicate decryption can be avoided
+        // TODO: This kind of optimization is memory intensive and could be an attack vector, how slow could things be if we ignore this?
```

Comment:
`valid_breaches` in itself could be an attack vector.
I could send the tower too many valid data that's triggered in block x and it will decrypt it and have it in memory.
I was gonna merge the part:
```rust
	for (uuid, user_id, breach) in valid_breaches {
            if let ConfirmationStatus::Rejected(_) =
                self.responder.handle_breach(uuid, breach, user_id)
            {
                invalid_breaches.push(uuid);
            }
        }
```
into `filter_breaches`. This way, as beaches are decrypted, we will handle them immediately without having them residing in memory.